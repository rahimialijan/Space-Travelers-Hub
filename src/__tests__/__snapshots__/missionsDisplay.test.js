import React from 'react';
import renderer from 'react-test-renderer';
import MissionsView from '../../components/Missions/missionsDisplay';

describe('MissionsView component', () => {
  it('should render correctly when loading', () => {
    const component = renderer.create(
      <MissionsView
        missions={[]}
        isLoading
        error={null}
        handleJoinMission={() => {}}
        handleLeaveMission={() => {}}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when there is an error', () => {
    const error = 'An error occurred.';
    const component = renderer.create(
      <MissionsView
        missions={[]}
        isLoading={false}
        error={error}
        handleJoinMission={() => {}}
        handleLeaveMission={() => {}}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with mission data', () => {
    const missions = [
      {
        mission_id: 1,
        mission_name: 'Mission 1',
        description: 'Description 1',
        reserved: true,
      },
      {
        mission_id: 2,
        mission_name: 'Mission 2',
        description: 'Description 2',
        reserved: false,
      },
    ];
    const component = renderer.create(
      <MissionsView
        missions={missions}
        isLoading={false}
        error={null}
        handleJoinMission={() => {}}
        handleLeaveMission={() => {}}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
