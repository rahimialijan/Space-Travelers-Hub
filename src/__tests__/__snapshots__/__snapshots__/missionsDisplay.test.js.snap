// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsView component should render correctly when loading 1`] = `
<div>
  Loading...
</div>
`;

exports[`MissionsView component should render correctly when there is an error 1`] = `
<div>
  Error:
  An error occurred.
</div>
`;

exports[`MissionsView component should render correctly with mission data 1`] = `
<div>
  <table
    className="striped-table"
  >
    <thead>
      <tr>
        <th
          className="nowrap"
        >
          Mission Name
        </th>
        <th
          className="nowrap"
        >
          Description
        </th>
        <th
          className="nowrap"
        >
          Status
        </th>
        <th
          className="nowrap"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="even-row"
      >
        <td
          className="bold"
        >
          Mission 1
        </td>
        <td>
          Description 1
        </td>
        <td
          className="button-cell"
        >
          <button
            className="active-member status-btn"
            type="button"
          >
            Active Member
          </button>
        </td>
        <td
          className="button-cell"
        >
          <button
            className="leave-mission action-btn"
            onClick={[Function]}
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
      <tr
        className="odd-row"
      >
        <td
          className="bold"
        >
          Mission 2
        </td>
        <td>
          Description 2
        </td>
        <td
          className="button-cell"
        >
          <button
            className="not-member status-btn"
            type="button"
          >
            Not Member
          </button>
        </td>
        <td
          className="button-cell"
        >
          <button
            className="join-mission action-btn"
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
